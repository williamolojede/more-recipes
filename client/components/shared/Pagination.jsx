import React from 'react';
import PropTypes from 'prop-types';

const Pagination = ({ currentPage, pages, getRecipesForPage }) => (
  <ul className="pagination">
    <li
      className={currentPage === 1 ? 'disabled' : 'waves-effect'}
      id="pagination__prev"
    >
      <a href="#!" onClick={() => getRecipesForPage(currentPage - 1)}>
        <i className="material-icons">chevron_left</i>
      </a>
    </li>
    {
      pages.map(page => (
        <li
          className={page === currentPage ? 'active' : 'waves-effect'}
          key={page}
        >
          <a href="#!" onClick={() => getRecipesForPage(page)}>
            {page}
          </a>
        </li>
      ))
    }
    <li
      className={
        currentPage === pages[pages.length - 1] ?
          'disabled' : 'waves-effect'
      }
      id="pagination__next"
    >
      <a href="#!" onClick={() => getRecipesForPage(currentPage + 1)}>
        <i className="material-icons">chevron_right</i>
      </a>
    </li>
  </ul>
);


Pagination.propTypes = {
  currentPage: PropTypes.number.isRequired,
  pages: PropTypes.arrayOf(PropTypes.number).isRequired,
  getRecipesForPage: PropTypes.func.isRequired
};

export default Pagination;
