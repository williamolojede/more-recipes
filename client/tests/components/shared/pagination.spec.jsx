import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '../../../components/shared/Pagination';

function setup() {
  const props = {
    currentPage: 1,
    pages: [1, 2, 3, 4, 5],
    getRecipesForPage: jest.fn()
  };
  const component = renderer.create(<Pagination {...props} />);
  const wrapper = shallow(<Pagination {...props} />);

  return {
    props,
    component,
    wrapper
  };
}

describe('Pagination', () => {
  it('should render with right amount of elements', () => {
    const tree = setup().component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('links', () => {
    it('should respond to click event', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
      const buttons = wrapper.find('li button');
      const getRecipesForPageSpy = jest.spyOn(wrapper.instance().props, 'getRecipesForPage');
      buttons.forEach(button => button.simulate('click'));
      expect(getRecipesForPageSpy).toHaveBeenCalledTimes(7);
    });
  });

  describe('previous link', () => {
    it('should not be disabled if currentpage is not page 1', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
      wrapper.setProps({ currentPage: 2 });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('next link', () => {
    it('should be diabled if current page is the last', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
      wrapper.setProps({ currentPage: 5 });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
