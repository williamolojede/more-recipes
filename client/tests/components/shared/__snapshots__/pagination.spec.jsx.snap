// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination links should respond to click event 1`] = `
<ul
  className="pagination"
>
  <li
    className="disabled"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="active"
    key="1"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="waves-effect"
    key="2"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
    key="3"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
    key="4"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="waves-effect"
    key="5"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="waves-effect"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;

exports[`Pagination next link should be diabled if current page is the last 1`] = `
<ul
  className="pagination"
>
  <li
    className="disabled"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="active"
    key="1"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="waves-effect"
    key="2"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
    key="3"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
    key="4"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="waves-effect"
    key="5"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="waves-effect"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;

exports[`Pagination next link should be diabled if current page is the last 2`] = `
<ul
  className="pagination"
>
  <li
    className="waves-effect"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="waves-effect"
    key="1"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="waves-effect"
    key="2"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
    key="3"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
    key="4"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="active"
    key="5"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="disabled"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;

exports[`Pagination previous link should not be disabled if currentpage is not page 1 1`] = `
<ul
  className="pagination"
>
  <li
    className="disabled"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="active"
    key="1"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="waves-effect"
    key="2"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
    key="3"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
    key="4"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="waves-effect"
    key="5"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="waves-effect"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;

exports[`Pagination previous link should not be disabled if currentpage is not page 1 2`] = `
<ul
  className="pagination"
>
  <li
    className="waves-effect"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="waves-effect"
    key="1"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="active"
    key="2"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
    key="3"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
    key="4"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="waves-effect"
    key="5"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="waves-effect"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;

exports[`Pagination should render with right amount of elements 1`] = `
<ul
  className="pagination"
>
  <li
    className="disabled"
    id="pagination__prev"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    </a>
  </li>
  <li
    className="active"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="waves-effect"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="waves-effect"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="waves-effect"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="waves-effect"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="waves-effect"
    id="pagination__next"
  >
    <a
      href="#!"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </a>
  </li>
</ul>
`;
